webui/ocw/lib/gce.py

```diff
@@ -30,35 +30,75 @@ def compute_client(self):
             self.__compute_clinet = googleapiclient.discovery.build('compute', 'v1', credentials=credentials)
         return self.__compute_clinet
 
-    def list_instances(self, zone='europe-west1-b'):
-        i = self.compute_client().instances().list(project=self.__project, zone=zone).execute()
-        return i['items'] if 'items' in i else []
+    def list_instances(self, zone):
+        result = []
+        request = self.compute_client().instances().list(project=self.__project, zone=zone)
+        while request is not None:
+            response = request.execute()
+            if 'items' in response:
```

Comment:
and if there is no 'items' in response ? maybe we need it least log this ?

Reply:
Don't know. Thins only means, that there  is no instance running at CSP. Which could happen.